wordpress-mobile/WordPress-iOS
# Remove the .show2FA segue and replace with programmatic nav

Fixes # n/a

This PR removes the segue that sends a user to the 2-factor authentication screen. It now navigates the user programmatically.

Ref. https://github.com/wordpress-mobile/WordPressAuthenticator-iOS/pull/228

## Requirements before testing
**The object of the game is to reach this screen 👇 through as many different methods as possible.**

<img src="https://user-images.githubusercontent.com/1062444/78379505-17067780-7598-11ea-87c2-3962ec03bf7a.PNG" width="350" />

- Have a real Apple device (no sims), signed into the testing account you'll use to test SIWA on the app
- Have [2-factor authentication enabled](https://wordpress.com/support/security/two-step-authentication/) enabled on a WP.com account that uses email. (This account should have a WP.com hosted site.)
- Have 2-factor authentication enabled on a WP.com account that signs in through Google Sign In
- Have 2-factor authentication enabled on a WP.com account that signs in through SIWA. Ensure you have created a free site to go along with the account.
- Have 2-factor authentication enabled on a WP.com account that signs in through Safari Saved Passwords
- Have 2-factor authentication enabled for a *.wp.com site address that can sign in via username / password
- ~Have a self-hosted site that has 2-factor authentication enabled (This is a Jetpack site with 2-factor auth enabled on your WP.com account AND on your site's individual settings.)~

### To test:
1. Ensure the pre-requisites are fulfilled
2. Check out this branch
3. `rake dependencies`
4. Build and run
5. Navigate through each of these flows:

#### Flow 1, Login via WP.com hosted site address + password, checks for 2fa
Flow 1 doesn't navigate to the 2fa VC. Instead, this flow checks that the other 2fa process was not broken during the segue removal changes.

1. Login > Continue with your site address
2. Use a WP.com site address that has 2fa enabled > Next (ex: thuycopeland.com)
3. Login Self-Hosted VC will appear. Fill in normal WP.com username / password > Next
4. An alert asking you to use an app password should appear. Dismiss by tapping OK button.
5. Visit [two-step authentication](https://wordpress.com/me/security/two-step) and generate an app-specific password.
5. Enter the app-specific password into the password field.
**Expected:** it navigates to the "Logged in as..." screen that has a Continue button.

#### Flow 2, Login via WP.com email + password
1. Login > Continue with WordPress.com > enter email address > Next
2. Enter password > Next > **expected:** the 2fa screen should appear
3. Enter 2fa code > Next
**Expected:** "Logged in as..." screen that has a Continue button should appear.

#### Flow 3, Google Sign In
1. Login > Continue with Google > Continue on the alert message
2. A modal window will appear. Follow Google's instructions to sign in
3. Once the Google Sign In flow completes, the modal window will dismiss. It may take a few seconds to navigate from the "carousel" screen to the 2FA screen. **expected:** the 2fa screen appears 
4. Use a verification code > Next
**Expected:** "Logged in as..." screen that has a Continue button should appear.

#### Flow 4, Sign in with Apple (SIWA)
1. Login > Continue with Apple > Continue
2. One the SIWA flow completes, the bottom sheet will dismiss. **expected:** the 2fa screen appears 
3. Use a verification code > Next
**Expected:** "Logged in as..." screen that has a Continue button should appear.

#### Flow 5, `*.wordpress.com` site, Login by Site Address
1. Login > "entering your site address" button text OR Login > Continue with WordPress.com > "entering your site address" button text
2. Type in the free site you set up previously, `*.wordpress.com` (ex: thuycopeland900625632.wordpress.com) > Next
3. Use the WP.com account username / password > Next. **expected: the 2fa screen appears**
4. Use a verification code > Next
**Expected:** "Logged in as..." screen that has a Continue button should appear.

## Do not merge until:
1. The authenticator PR is merged https://github.com/wordpress-mobile/WordPressAuthenticator-iOS/pull/228 ✔️
2. The latest Authenticator release is published to CocoaPods trunk  ✔️
3. This branch is updated to point to the latest Auth release  ✔️

PR submission checklist:
